tasks/configure.yml

```diff
@@ -27,3 +17,36 @@
   when:
     - ansible_version.full is version_compare('2.4', '>=')
     - ansible_selinux.status == "enabled"
+
+- block:
+    - name: Create textfile collector dir
+      file:
+        path: "{{ node_exporter_textfile_dir }}"
+        state: directory
+        owner: "{{ node_exporter_system_user }}"
+        group: "{{ node_exporter_system_group }}"
+        mode: 0775
+
+    - name: Download textfile collector scripts
+      copy:
+        src: "{{ item.src }}"
+        dest: "{{ _node_exporter_binary_install_dir }}/{{ item.src | basename }}"
+        owner: root
+        group: root
+        mode: "0755"
+      with_items: "{{ node_exporter_textfile_collectors }}"
+
+    - name: Setup textfile collector script cronjob
+      cron:
+        cron_file: "/etc/crontab"
+        name: "textfile collector {{ item.src | basename }}"
+        user: "{{ item.user | default('root') }}"
+        minute: "{{ item.minute | default(omit) }}"
+        hour: "{{ item.hour | default(omit) }}"
+        day: "{{ item.day | default(omit) }}"
+        month: "{{ item.month | default(omit) }}"
+        weekday: "{{ item.weekday | default(omit) }}"
+        special_time: "{{ item.special_time | default(omit) }}"
+        job: "{{ _node_exporter_binary_install_dir }}/{{ item.src | basename }} | sponge {{ node_exporter_textfile_dir }}/{{ item.src | basename }}.prom"
```

Comment:
We need to make sure the `moreutils` package is installed for sponge.

Reply:
Good point! :+1:

Reply:
And of course `moreutils` is not available in main CentOS repositories. Is there a way of doing this without `sponge` or should I just add `epel-release` repo when running on CentOS?

Reply:
Hrm, I don't know of a good sponge replacement that would be built-in to most distros.

Reply:
I found what `sponge` is doing and it is achievable to do the same with basic linux tools.
Basically it writes what it receives on stdin to a temporary file and then renames temporary file to a target one. Which means we can do the same with `mktemp && mv`. I'll create a POC during the weekend.